### PR TITLE
[ML] add nlp config update serialization tests

### PR DESCRIPTION
**Pull request by benwtrent**
adds inference request serialization tests that include nlp config updates.

relates: https://github.com/elastic/elasticsearch/pull/85863

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/InternalInferModelActionRequestTests.java [hunk 1]
```diff
@@ -68,6 +87,27 @@ protected NamedWriteableRegistry getNamedWriteableRegistry() {
 
     @Override
     protected Request mutateInstanceForVersion(Request instance, Version version) {
-        return instance;
+        InferenceConfigUpdate adjustedUpdate;
+        InferenceConfigUpdate currentUpdate = instance.getUpdate();
+        if (currentUpdate instanceof NlpConfigUpdate nlpConfigUpdate) {
```
Comment (davidkyle) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/InternalInferModelActionRequestTests.java:92]: Could you use a fancy instanceof switch statement here?

```
switch (currentUpdate) {
        case NlpConfigUpdate i -> NlpConfigUpdateTests.mutateForVersion(update, version);
        ....
};
```

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
@elasticmachine update branch

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/InternalInferModelActionRequestTests.java (hunk 1 above)
Comment (benwtrent) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/InternalInferModelActionRequestTests.java]: oh snap, you are correct!

**Pull request merged**